docs/tutorials/text_prediction/automm_customization.md

```diff
@@ -0,0 +1,236 @@
+```
+
+## Environment
+
+### env.num_gpus
+The number of gpus to use.
+```
+# by default, all available gpus are used by AutoMM
+predictor.fit(hyperparameters={"env.num_gpus": -1})
+# use 1 gpu only
+predictor.fit(hyperparameters={"env.num_gpus": 1})
+```
+
+### env.per_gpu_batch_size
+The batch size for each GPU. 
+```
+# default used by AutoMM
+predictor.fit(hyperparameters={"env.per_gpu_batch_size": 8})
+# use batch size 16 per GPU
+predictor.fit(hyperparameters={"env.per_gpu_batch_size": 16})
+```
+
+### env.batch_size
+The batch size to use in each step of training. If `env.batch_size` is larger than `env.per_gpu_batch_size * env.num_gpus`, we accumulate gradients to reach the effective `env.batch_size` before performing one optimization step.
+```
+# default used by AutoMM
+predictor.fit(hyperparameters={"env.batch_size": 128})
+# use batch size 256
+predictor.fit(hyperparameters={"env.batch_size": 256})
+```
+
+### env.eval_batch_size_ratio
+Prediction or evaluation uses a larger per gpu batch size `env.per_gpu_batch_size * env.eval_batch_size_ratio`.
+```
+# default used by AutoMM
+predictor.fit(hyperparameters={"env.eval_batch_size_ratio": 4})
+# use 2x per gpu batch size during prediction or evalution
+predictor.fit(hyperparameters={"env.eval_batch_size_ratio": 2})
+```
+
+### env.precision
+Support either double (`64`), float (`32`), bfloat16 (`"bf16"`), or half (`16`) precision training.
+
+Half precision, or mixed precision, is the combined use of 32 and 16 bit floating points to reduce memory footprint during model training. This can result in improved performance, achieving +3X speedups on modern GPUs.
+```
+# default used by AutoMM
+predictor.fit(hyperparameters={"env.precision": 16})
+# use bfloat16
+predictor.fit(hyperparameters={"env.precision": "bf16"})
+```
+
+### env.num_workers
+The number of worker processes used by Pytorch dataloader in training.
+
+```
+# default used by AutoMM
+predictor.fit(hyperparameters={"env.num_workers": 2})
+# use 4 workers in the training dataloader
+predictor.fit(hyperparameters={"env.num_workers": 4})
+```
+
+### env.num_workers_evaluation
+The number of worker processes used by Pytorch dataloader in prediction or evaluation.
+
+```
+# default used by AutoMM
+predictor.fit(hyperparameters={"env.num_workers_evaluation": 2})
+# use 4 workers in the prediction/evaluation dataloader
+predictor.fit(hyperparameters={"env.num_workers_evaluation": 4})
+```
+
+### env.strategy
+Distributed training mode.
+- `"dp"`: data parallel.
+- `"ddp"`: distributed data parallel (python script based).
+- `"ddp_spawn"`: distributed data parallel (spawn based).
+
+See [here](https://pytorch-lightning.readthedocs.io/en/stable/accelerators/gpu.html#distributed-modes) for more details.
+```
+# default used by AutoMM
+predictor.fit(hyperparameters={"env.strategy": "ddp_spawn"})
+# use ddp during training
+predictor.fit(hyperparameters={"env.strategy": "ddp"})
+```
+
+## Model
```

Comment:
Are we leaving `model` and `data` for next PR or they have too much content to be put in a tutorial?

Reply:
Just added.